internal/consensus/state.go

```diff
@@ -1506,6 +1506,21 @@ func (cs *State) defaultDoPrevote(ctx context.Context, height int64, round int32
 		}
 	}
 
+	stateMachineValidBlock, err := cs.blockExec.ProcessProposal(ctx, cs.ProposalBlock)
```

Comment:
Can we add a comment here explaining what is happening

Reply:
Agreed, and also added a warning about misusing block rejection